debug-test/src/main.rs

```diff
@@ -0,0 +1,106 @@
+// SPDX-FileCopyrightText: 2022 Google LLC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#![no_std]
+#![cfg_attr(not(test), no_main)]
+
+use core::fmt::Write;
+
+use heapless::String;
+#[cfg(not(test))]
+use riscv_rt::entry;
+
+#[cfg(not(test))]
+extern crate panic_halt;
+
+const ADDR: *mut u8 = 0x0000_1000 as *mut u8;
+
+fn print(s: &str) {
+    for b in s.bytes() {
+        unsafe {
+            ADDR.write_volatile(b);
+        }
+    }
+}
+
+#[cfg_attr(not(test), entry)]
+fn main() -> ! {
+    print("hello, world.\n");
+
+    print("I am here to be debugged!\n");
+
+    loop {
+        for i in 0..30 {
+            let mut s = String::<16>::new();
+            let _ = writeln!(s, "Hey! {i}");
```

Comment:
This is using `core::fmt` instead of `ufmt`, which I guess is fine since this is just for simulation and we don't really have the same restrictions with memory sizes, but maybe we should still use `ufmt` here?

Reply:
I don't think it makes too much sense to do it here, to really get the size benefits we need a nightly Rust toolchain, for debugging purposes this is just fine